projects/components/src/stepper/stepper.component.ts

```diff
@@ -138,14 +167,32 @@ export class StepperComponent implements AfterContentInit {
     }
   }
 
+  public nextOrSubmit(stepper: MatStepper): void {
+    const isLastStep = stepper.selectedIndex === stepper.steps.length - 1;
+    isLastStep ? this.submitted.emit() : stepper.next();
+  }
+
   public getActionButtonLabel(selectedIndex: number, steps: StepperTabComponent[]): string {
-    return steps[selectedIndex]?.actionButtonLabel ?? (selectedIndex === steps.length ? 'Submit' : 'Next');
+    const isLastStep = selectedIndex === steps.length - 1;
+
+    return steps[selectedIndex]?.actionButtonLabel ?? (isLastStep ? 'Submit' : 'Next');
   }
 
   private isStepCompleted(index: number): boolean {
     const step = this.steps.get(index);
+    if (!step) {
+      return false;
+    }
+
+    return step.stepControl?.valid ?? step.completed;
+  }
 
-    return step?.completed ?? false;
+  /**
+   * Check if all the steps are valid.
+   * If the step has a step control , we check the forms validity and if not, we check the step completed property.
+   */
+  private areAllStepsValid(): boolean {
+    return this.steps.toArray().every(step => step.stepControl?.valid ?? step.completed);
```

Comment:
you can use the above defined `isStepCompleted` method here.

Reply:
Updated.